src/cmd/ksh93/edit/emacs.c

```diff
@@ -1002,7 +1004,13 @@ static int escape(register Emacs_t* ep,register genchar *out,int count)
 		case '*':		/* filename expansion */
 		case '=':	/* escape = - list all matching file names */
 			ep->mark = cur;
-			if(cur<1)
+			for(x=0; x <= cur; x++)
+				if(out[x] > ' ')
```

Comment:
This is ASCII-centric. This code base still theoretically supports EBCDIC (though I've been unable to test it). Also, not all control characters are whitespace. Better would be: `if(!isspace(out[x]))`